### PR TITLE
[query] Respect M3Cluster headers in Namespace Get

### PR DESCRIPTION
**Pull request by schallert**
**What this PR does / why we need it**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note
Ensure Cluster-*-Name headers are respected in the m3admin namespace get handler
```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note

```

### CODE REVIEW

File: src/query/api/v1/handler/namespace/get.go [hunk 1]
```diff
@@ -71,10 +72,15 @@ func NewGetHandler(
 	}
 }
 
-func (h *GetHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+func (h *GetHandler) ServeHTTP(
+	svc handleroptions.ServiceNameAndDefaults,
```
Comment (arnikola) [inline src/query/api/v1/handler/namespace/get.go:76]: Doesn't `GetHandler` need to implement `http.Handler`?

File: src/x/debug/namespace.go [hunk 2]
```diff
@@ -36,25 +38,46 @@ import (
...
+	handler  *namespace.GetHandler
+	defaults handleroptions.ServiceNameAndDefaults
 }
 
 // NewNamespaceInfoSource returns a Source for namespace information.
 func NewNamespaceInfoSource(
 	clusterClient clusterclient.Client,
 	instrumentOpts instrument.Options,
-) Source {
+	allDefaults []handleroptions.ServiceNameAndDefaults,
```
Comment (arnikola) [inline src/x/debug/namespace.go:49]: nit: opts as last arg

File: src/query/api/v1/handler/namespace/get.go (hunk 1 above)
Comment (schallert) [reply src/query/api/v1/handler/namespace/get.go:76]: The convention we use in the other namespace handlers is to have this signature and wrap them, so just pattern matching that: https://github.com/m3db/m3/blob/2cc61a79909d6b3e18ffab35846a2e428334b637/src/query/api/v1/handler/namespace/common.go#L125-L155.

Comment (schallert) [reply src/query/api/v1/handler/namespace/get.go:76]: I just realized though that the `Delete` handler should be updated to be wrapped as well, will add that on.

**Pull request merged**